source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -70,11 +72,59 @@ namespace aspect
       double
       CompositeViscoPlastic<dim>::compute_viscosity (const double pressure,
                                                      const double temperature,
-                                                     const unsigned int composition,
+                                                     const std::vector<double> &volume_fractions,
                                                      const SymmetricTensor<2,dim> &strain_rate,
                                                      std::vector<double> &partial_strain_rates,
                                                      const std::vector<double> &phase_function_values,
                                                      const std::vector<unsigned int> &n_phases_per_composition) const
+      {
+        double viscosity = 0.;
+        partial_strain_rates.resize(5, 0.);
```

Comment:
What is the initial size of this vector that the function receives? Is it only just a single value (background strain rate) that is used as the initial guess?

Reply:
The vector could be any size, with values equal to zero. The resizing seemed like the safest option in case an empty vector was passed, rather than a vector of length 5. The alternative would be to check `(size == 5)`, but I'm don't think that would be significantly faster.